core/auth/interfaces/controller.go

```diff
@@ -0,0 +1,48 @@
+package interfaces
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	Authcontroller struct {
+		responder *web.Responder
+		authservice Authservice
+		router         web.ReverseRouter
+	}
+)
+
+
+// Inject for Authcontroller
+func (c *Authcontroller) Inject(responder *web.Responder, authservice Authservice,router web.ReverseRouter,) {
+	c.responder = responder
+	c.authservice = authservice
+	c.router = router
+}
+
+//AuthAction - default auth action - starting the authorsation with the registered Authservice
+func (c *Authcontroller) AuthAction(ctx context.Context, r *web.Request) web.Result {
+	redirecturl, ok := r.Params["redirecturl"]
+	if !ok || redirecturl == "" {
+		redirecturl = r.Request().Referer()
+	}
+
+	if refURL, err := url.Parse(redirecturl); err != nil || refURL.Host != r.Request().Host {
```

Comment:
I'm not sure should we have this logic here: there is same functionality in security middleware, so we can either move that here completely or we just read "redirecturl" Param and pass url or nil as argument.